ml_genn/model.py

```diff
@@ -234,6 +234,7 @@ def evaluate(self, data, labels, time, save_samples=[]):
                 for output_i in range(len(self.outputs)):
                     predictions = self.outputs[output_i].neurons.get_predictions(
                         pipe_batch_end - pipe_batch_start)
+                    batch_labels = [np.argmax(i) for i in batch_labels] if (batch_labels[0].shape != predictions[0].shape) else batch_labels
```

Comment:
It might be a little more readable (we _try_ and avoid really long lines) if you did the test on the shapes outside the loop at the top of the function like:
```python
category_labels = (batch_labels[0].shape != predictions[0].shape)
```
then you can simplify this line to:
```python
batch_labels = [np.argmax(i) for i in batch_labels] if category_labels else batch_labels
```

Reply:
Thanks, yes, it's more readable, it's in the new commit.